packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -715,7 +750,19 @@ export const MenuBarMixin = (superClass) =>
      */
     _setFocused(focused) {
       if (focused) {
-        const target = this.querySelector('[tabindex="0"]');
+        if (this.tabNavigation) {
+          // manage the submenu
+          const target = this.querySelector('[focused]');
+          const wasExpanded = this._expandedButton !== undefined && this._expandedButton !== target;
```

Comment:
You should use `this._expandedButton != undefined` to allow `null` because we set it to `null` when closing the sub-menu. Otherwise, `wasExpanded` will incorrectly return `true`.

Reply:
I've added the condition so it's working on the root menu items